Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/ReduceCognitiveLoadAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class ReduceCognitiveLoadAnalyzer : DiagnosticAnalyzer
+	{
+		public ReduceCognitiveLoadAnalyzer()
+		{ }
+
+		public ReduceCognitiveLoadAnalyzer(int maxCognitiveLoad)
+		{
+			MaxCognitiveLoad = maxCognitiveLoad;
+		}
+
+		private int MaxCognitiveLoad { get; set; }
+
+		private const string Title = @"Reduce Cognitive Load";
+		public const string MessageFormat = @"Reduce Cognitive Load of {0} to threshold of {1}";
+		private const string Description = @"This method has too many nested block statements and logical cases to consider in this method";
+		private const string Category = Categories.Maintainability;
+
+		private static readonly DiagnosticDescriptor Rule = new(Helper.ToDiagnosticId(DiagnosticIds.ReduceCognitiveLoad), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+			context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.MethodDeclaration);
+		}
+
+		private int CalcCognitiveLoad(BlockSyntax blockSyntax)
+		{
+			int cognitiveLoad = 1;
+			foreach (BlockSyntax descBlockSyntax in blockSyntax.DescendantNodes().OfType<BlockSyntax>())
+			{
+				cognitiveLoad += CalcCognitiveLoad(descBlockSyntax);
+			}
+			return cognitiveLoad;
+		}
+
+		private void Analyze(SyntaxNodeAnalysisContext context)
+		{
+			MethodDeclarationSyntax methodDeclarationSyntax = (MethodDeclarationSyntax)context.Node;
```

Comment:
Please add the GeneratedCodeDetector here

Reply:
Done